### PR TITLE
Remove `.spec.kubernetes.kubeAPIServer.enableBasicAuthentication` field from the Shoot API

### PR DESCRIPTION
**Pull request by dimitar-kostadinov**
**How to categorize this PR?**

/area control-plane
/kind api-change
/kind cleanup

**What this PR does / why we need it**:
This PR removes `.spec.kubernetes.kubeAPIServer.enableBasicAuthentication` after it has been deprecated in https://github.com/gardener/gardener/pull/6987. For existing Shoots the field was dropped (set to nil) with https://github.com/gardener/gardener/pull/7534.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/6911

**Special notes for your reviewer**:
N/A

**Release note**:

```breaking user
⚠️ The deprecated field `.spec.kubernetes.kubeAPIServer.enableBasicAuthentication` has been removed from the Shoot API. Please check your `Shoot`s manifests and remove the `.spec.kubernetes.kubeAPIServer.enableBasicAuthentication` field.
```

### CODE REVIEW

**Review by shafeeqes — COMMENTED** [review]
/lgtm

File: pkg/registry/core/shoot/strategy.go [hunk 1]
```diff
@@ -190,13 +189,6 @@ func cleanupAdmissionPlugins(shoot *core.Shoot) {
...
 
-// dropEnableBasicAuthenticationField sets the enableBasicAuthentication to nil.
-func dropEnableBasicAuthenticationField(shoot *core.Shoot) {
-	if shoot.Spec.Kubernetes.KubeAPIServer != nil {
-		shoot.Spec.Kubernetes.KubeAPIServer.EnableBasicAuthentication = nil
-	}
-}
-
 // setKubernetesDashboardAuthMode sets the kubernetes-dashboard authentication mode to "token", if its current value is not "token" (for example "basic").
 func setKubernetesDashboardAuthMode(shoot *core.Shoot) {
```
Comment (ialidzhikov) [inline pkg/registry/core/shoot/strategy.go]: Having a second look, let's also drop the `setKubernetesDashboardAuthMode` func. It was a migration code that did its job.

**Review by ialidzhikov — COMMENTED** [review]
Minor suggestion, otherwise lgtm.

File: pkg/registry/core/shoot/strategy_test.go [hunk 2]
```diff
@@ -576,26 +576,6 @@ var _ = Describe("Strategy", func() {
...
-			It("should do nothing when kubeAPIServer is nil", func() {
-				shoot.Spec.Kubernetes.KubeAPIServer = nil
-
-				shootregistry.NewStrategy(0).Canonicalize(shoot)
-
-				Expect(shoot.Spec.Kubernetes.KubeAPIServer).To(BeNil())
-			})
-		})
-
 		DescribeTable("kubernetesDashboard.authenticationMode field",
```
Comment (ialidzhikov) [inline pkg/registry/core/shoot/strategy_test.go]: You should remove the whole `DescribeTable`.

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**